### ISSUE TITLE
Flaky test - ErrorSquiggleProducerTests.ErrorDoesNotCrashPastEOF 

### ISSUE THREAD

**Issue by allisonchou** [dotnet/roslyn#46950]
Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Squiggles.ErrorSquiggleProducerTests.ErrorDoesNotCrashPastEOF

PR: #46946
Build: https://dev.azure.com/dnceng/public/_build/results?buildId=777134&view=ms.vss-test-web.build-test-results-tab&runId=24314366&resultId=120403&paneView=debug

Stack trace:
```
Assert.Equal() Failure\r\nExpected: 3\r\nActual:   0
 en Xunit.Assert.Equal[T](T expected, T actual, IEqualityComparer`1 comparer) en C:\projects\xunit\src\xunit.assert\Asserts\EqualityAsserts.cs:línea 40
   en Xunit.Assert.Equal[T](T expected, T actual) en C:\projects\xunit\src\xunit.assert\Asserts\EqualityAsserts.cs:línea 24
   en Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Squiggles.ErrorSquiggleProducerTests.<ErrorDoesNotCrashPastEOF>d__4.MoveNext() en /_/src/EditorFeatures/CSharpTest/Squiggles/ErrorSquiggleProducerTests.cs:línea 201
--- Fin del seguimiento de la pila de la ubicación anterior donde se produjo la excepción ---
   en System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   en System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   en Xunit.Sdk.TestInvoker`1.<>c__DisplayClass48_1.<<InvokeTestMethodAsync>b__1>d.MoveNext() en C:\projects\xunit\src\xunit.execution\Sdk\Frameworks\Runners\TestInvoker.cs:línea 264
--- Fin del seguimiento de la pila de la ubicación anterior donde se produjo la excepción ---
   en System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   en System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   en Xunit.Sdk.ExecutionTimer.<AggregateAsync>d__4.MoveNext() en C:\projects\xunit\src\xunit.execution\Sdk\Frameworks\ExecutionTimer.cs:línea 48
--- Fin del seguimiento de la pila de la ubicación anterior donde se produjo la excepción ---
   en System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   en System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   en Xunit.Sdk.ExceptionAggregator.<RunAsync>d__9.MoveNext() en C:\projects\xunit\src\xunit.core\Sdk\ExceptionAggregator.cs:línea 90
```

**Comment by allisonchou** [comment]
This may be the same as #46786.
Similar failure hit again here:
https://dev.azure.com/dnceng/public/_build/results?buildId=777326&view=ms.vss-test-web.build-test-results-tab&runId=24326910&resultId=120790&paneView=debug

**Comment by sharwell** [comment]
Reopening this one. The failure scenario looked identical, but we couldn't prove that my change to fix #46786 also fixes this.

**Comment by sharwell** [comment]
@mavasani So far I've traced the missing tags scenario back to a failure to call `DiagnosticIncrementalAnalyzer.RaiseDocumentDiagnosticsIfNeeded` with a non-empty `newItems`. I'm now working to further narrow down the problem to either a failure of `PersistAndRaiseDiagnosticsIfNeeded`, or a failure of `ComputeDocumentAnalysisDataAsync`.
